### PR TITLE
Prompt new users to customize their feed

### PR DESCRIPTION
**Pull request by maestromac**
## What type of PR is this? (check all applicable)
- [x] Feature

## Related Tickets & Documents
Resolves https://github.com/thepracticaldev/dev.to/issues/6706
## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
n/a
## Added tests?
- [x] yes

## Added to documentation?
- [x] no documentation needed

## [optional] Are there any post deployment tasks we need to perform?
- [x] Create new broadcast on production.

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -250,6 +250,7 @@
   welcome_thread: "Sloan here again! 👋 DEV is a friendly community. Why not introduce yourself by leaving a comment in <a href='/welcome'>the welcome thread</a>!",
   twitter_connect: "You're on a roll! 🎉 Let's connect your <a href='/settings'> Twitter account</a> to complete your identity so that we don't think you're a robot. 🤖",
   github_connect: "You're on a roll! 🎉 Let's connect your <a href='/settings'> GitHub account</a> to complete your identity so that we don't think you're a robot. 🤖",
+  customize_feed: "Hi, it's me again! 👋Now that you're a part of the DEV community, let's focus on personalizing your content. You can start by <a href='/tags'> following some tags </a> to help customize your feed! 🎉",
```
Comment (mstruve) [inline db/seeds.rb]: Looks like our convention is a space after the emoji, we should probably do that here as well.

File: app/services/broadcasts/welcome_notification/generator.rb [hunk 2]
```diff
@@ -57,12 +64,20 @@ def authenticated_with_all_providers?
...
       end
 
-      def customize_broadcast
-        @customize_broadcast ||= Broadcast.find_by(title: "Welcome Notification: customize_experience")
+      def customize_ux_broadcast
+        @customize_ux_broadcast ||= Broadcast.find_by(title: "Welcome Notification: customize_experience")
+      end
+
+      def customize_feed_broadcast
+        @customize_feed_broadcast ||= Broadcast.find_by(title: "Welcome Notification: customize_feed")
```
Comment (mstruve) [inline app/services/broadcasts/welcome_notification/generator.rb]: What is the plan for creating this new Broadcast in production?

Comment (vaidehijoshi) [reply app/services/broadcasts/welcome_notification/generator.rb]: @mstruve Currently, we are adding them via the `/internal/broadcasts` page as part of the deployment process. Notably, none of this stuff is live to users. But we are adding them as we merge.

Ultimately, however, we want to make these broadcasts more flexible and modifiable by admins for different communities! We are still working out, as a team, the way that we want to do that. Down the road, we likely will be using the data update scripts to pre-populate some generic Broadcasts for the welcome notification flow, and then allow admins of different communities to turn this feature off/on explicitly.

Comment (mstruve) [reply app/services/broadcasts/welcome_notification/generator.rb]: Excellent, just wanted to make sure we were thinking about all of that stuff. Thanks for the great explanation!

File: spec/services/broadcasts/welcome_notification/generator_spec.rb [hunk 3]
```diff
@@ -136,4 +137,40 @@
...
+      expect(Notification).not_to have_received(:send_welcome_notification)
+    end
+
+    it "does not send a notification to a user that is following 2 tags" do
+      2.times { user.follow(create(:tag)) }
+      sidekiq_perform_enqueued_jobs { described_class.new(user.id).send(:send_feed_customization_notification) }
+      expect(Notification).not_to have_received(:send_welcome_notification)
+    end
+
+    it "does send a notification to a user with 0 tag follows created 3 days ago" do
```
Comment (vaidehijoshi) [inline spec/services/broadcasts/welcome_notification/generator_spec.rb]:
```suggestion
    it "sends a notification to a user with 0 tag follows" do
```

does the created at 3 days ago really make a difference here?

File: db/seeds.rb (hunk 1 above)
Comment (maestromac) [reply db/seeds.rb]: Nice catch!
```suggestion
  customize_feed: "Hi, it's me again! 👋 Now that you're a part of the DEV community, let's focus on personalizing your content. You can start by <a href='/tags'> following some tags </a> to help customize your feed! 🎉",
```

File: app/services/broadcasts/welcome_notification/generator.rb [hunk 4]
```diff
@@ -38,12 +39,18 @@ def send_authentication_notification
...
-        return if received_notification?(customize_broadcast) || user.created_at > 5.days.ago
+        return if received_notification?(customize_ux_broadcast) || user.created_at > 5.days.ago
 
-        Notification.send_welcome_notification(user.id, customize_broadcast.id)
+        Notification.send_welcome_notification(user.id, customize_ux_broadcast.id)
         @notification_enqueued = true
       end
 
+      def send_feed_customization_notification
+        return if user_is_following_tags? || received_notification?(customize_feed_broadcast) || user.created_at > 3.days.ago
```
[the remark this answers is not in the document]
Comment (maestromac) [reply app/services/broadcasts/welcome_notification/generator.rb]: I don't have strong opinion on this either. It's like this here we are accessing cache (`user.cached_followed_tag_names`) so it's faster and I'm also assuming most users would follow some tag during their onboarding.

[the remark this answers is not in the document]
Comment (vaidehijoshi) [reply app/services/broadcasts/welcome_notification/generator.rb]: Sure, I think that makes sense. I don't think this is a blocker by any means, just wondering that while I was looking through the code.

**Pull request merged**